### PR TITLE
Submission dispatcher class refactor

### LINKED ISSUE #74: Refactor SubmissionDispatcher class
what is a `processingPhaseTag2`. In fact, we need to add *good* docs related to this tag and digest stuff. I mean, with some examples. This is non-trivial stuff (so, need proper doc).

As I commented before, the code you added looks good to me but we need to refactor legacy code:

1. remove silly documentation (e.g. "This function adds new user." in a method called `addNewUserInCatalog` does not add much, so remove the setence)
2. remove dead code
3. overall, we have important code in classes name *Util (e.g. CatalogUtils) and ancillary code in the main classes (e.g. formatPathRow)

_Originally posted by @thiagomanel in https://github.com/ufcg-lsd/saps-engine/pull/59_

### PR DESCRIPTION
**Pull request by thiagoyeds**
Fix #74

### CODE REVIEW

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java [hunk 1]
```diff
@@ -31,14 +30,8 @@ SapsImage addImageTask(String taskId, String dataset, String region, Date date,
 	void addUser(String userEmail, String userName, String userPass, boolean userState, boolean userNotify,
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: That name `addStateStamp` doesn't look good to me. Would something like `setStateChangeTime` be better?

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/DatabaseApplication.java [hunk 2]
```diff
@@ -224,7 +224,7 @@ public Restlet createInboundRoot() {
 	 * @throws SQLException
 	 */
 	public List<SapsImage> getTasksInState(ImageTaskState state) throws SQLException {
-		return this.submissionDispatcher.getTasksInState(state);
+		return this.submissionDispatcher.getTasksWithStateInCatalog(state);
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/DatabaseApplication.java]: Replace `With` by `By`

**Review by wesleymonte — CHANGES_REQUESTED** [review]

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java (hunk 1 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: `setStateChange` seems to me that there will be an update somewhere, but in fact it is an insertion of new information in the Catalog about a change of state of a task.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/DatabaseApplication.java (hunk 2 above)
Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/restlet/DatabaseApplication.java]: Done

File: src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java (hunk 1 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: So could you replace it with `addStateChangeTime`?

Comment (thiagoyeds) [reply src/main/java/org/fogbowcloud/saps/engine/core/database/Catalog.java]: lgtm @wesleymonte, done

**Review by wesleymonte — APPROVED** [review]

**Comment by thiagoyeds** [comment]
This PR is making changes to the same code base as PR #76, so it will now be closed and problem #74 will be resolved after PR #76 is accepted and merged.

**Comment by thiagoyeds** [comment]
This PR was closed because it did not request merging with the branch develop.

**Pull request closed without merging**